### PR TITLE
FW/logging: split the crash / OS error result from the result: line

### PR DESCRIPTION
**Pull request by thiagomacieira**
The other result types were always a string, so some parsers failed to
properly extract the details here and ended up blank.

[ChangeLog][Framework] Changed how the YAML output format logs crashes
and other unexpected operating system errors. Previously, they were a
YAML object in the "results" entry, instead of the "pass" or "fail"
string. Now, those will print either "crash" or "operating system error"
for that entry, with the details moved to the "result-details" entry.

Signed-off-by: Thiago Macieira <thiago.macieira@intel.com>

### CODE REVIEW

File: framework/logging.cpp [hunk 1]
```diff
@@ -1923,7 +1923,7 @@ std::string TapFormatLogger::fail_info_details()
     case STATUS_STACK_BUFFER_OVERRUN:
         return "Stack buffer overrun";
     }
-    return nullptr;
+    return "unknown";
```
Comment (busykai) [inline framework/logging.cpp:1926]: Could you capitalize this?

File: framework/logging.cpp [hunk 2]
```diff
@@ -1948,6 +1949,7 @@ std::string TapFormatLogger::fail_info_details()
     case EX_PROTOCOL:   return "remote error in protocol";
     case EX_NOPERM:     return "permission denied";
     case EX_CONFIG:     return "configuration error";
+    case EXIT_MEMORY:   return "Failed to perform an action due to memory shortage.";
```
Comment (busykai) [inline framework/logging.cpp:1952]: Strings are capitalized inconsistently (new ones are, the existing are not).

File: framework/simd.conf [hunk 3]
```diff
@@ -169,7 +169,7 @@ arch=CNL    	PLC
 arch=ICL	SNC
 arch=TGL	WLC
 arch=ADL	GRT
-#arch=RPL	GRT
+arch=RPL	GRT
```
Comment (busykai) [inline framework/simd.conf:172]: Why this change is part of this commit?

File: framework/logging.cpp (hunk 2 above)
Comment (thiagomacieira) [reply framework/logging.cpp:1952]: They come from different sources... Fixed.

**Review by busykai — APPROVED** [review]
Look good!
